File: laboratory/src/main/java/org/sopt/laboratory/lab8/member/service/MemberService.java
package org.sopt.laboratory.lab8.member.service;


import lombok.RequiredArgsConstructor;
import org.sopt.laboratory.lab8.common.exception.enums.ErrorType;
import org.sopt.laboratory.lab8.common.exception.model.NotFoundException;
import org.sopt.laboratory.lab8.member.domain.Member;
import org.sopt.laboratory.lab8.member.repository.MemberJpaRepository;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
@Transactional
public class MemberService {
    private final MemberJpaRepository memberJpaRepository;

    public Member getMemberById(final Long memberId) {
        return memberJpaRepository.findById(memberId)
                .orElseThrow(() -> new NotFoundException(ErrorType.MEMBER_NOT_FOUND_EXCEPTION));
    }
}

File: sse-kt/src/main/kotlin/org/practice/sse/service/MemberService.kt
package org.practice.sse.service

import org.practice.sse.controller.dto.UserCreate
import org.practice.sse.domain.Member
import org.practice.sse.exception.NotFoundException
import org.practice.sse.repository.MemberRepository
import org.springframework.stereotype.Service


@Service
class MemberService (private val memberRepository: MemberRepository) {

    fun createUser(userCreate: UserCreate) {
        val user = Member.of(userCreate)
        memberRepository.save(user)
    }

    fun findById(id: Long): Member {
        return memberRepository.findById(id) ?: throw NotFoundException("유저가 없습니다.")
    }
}